website/docs/plugin/framework/deprecations.mdx

```diff
@@ -0,0 +1,1282 @@
+
+		Attributes: map[string]schema.Attribute{
+			// ... other attributes ...
+		},
+		DeprecationMessage: "use example_thing resource instead",
+	}
+}
+
+func (e *exampleWidgetResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	resp.Diagnostics.Append(
+		diag.NewErrorDiagnostic("example_widget resource deprecated", "use example_thing resource instead"),
+	)
+}
+
+func (e *exampleWidgetResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	resp.Diagnostics.Append(
+		diag.NewErrorDiagnostic("example_widget resource deprecated", "use example_thing resource instead"),
+	)
+}
+
+func (e *exampleWidgetResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.Append(
+		diag.NewErrorDiagnostic("example_widget resource deprecated", "use example_thing resource instead"),
+	)
+}
+
+func (e *exampleWidgetResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	resp.Diagnostics.Append(
+		diag.NewErrorDiagnostic("example_widget resource deprecated", "use example_thing resource instead"),
+	)
+}
+
+```
+
+To remove `example_widget`:
+```go
+// ... provider implementation ...
+
+func (p *exampleProvider) Resources(ctx context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		//... other resources ...
+	}
+}
+```
+
+## Provider Data Source or Resource Rename
+
+When renaming a resource from one name to another, it is important to keep backwards compatibility with both existing Terraform configurations and the Terraform state while operators migrate. To accomplish this, there will be some duplicated logic to support both resources until the next `MAJOR` release. Once both resources are appropriately handled, the process for deprecating and removing the old resource is the same as noted in the [Provider Data Source or Resource Removal section](#provider-data-source-or-resource-removal).
+
+The recommended process is as follows:
+
+1. Duplicate the code of the existing resource, renaming (and potentially modifying) functions as necessary.
+1. Duplicate the documentation of the existing resource, renaming (and potentially modifying) as necessary.
+1. Add a [`DeprecationMessage`](/terraform/plugin/framework/handling-data/schemas#deprecationmessage) to the schema definition of the existing (now the "old") resource, noting to use the new resource in the message.
+1. Add `~> This resource is deprecated and will be removed in the next major version` to the documentation of the existing (now the "old") resource, noting to use the new resource.
+1. Add the new resource to the provider [`Resources`](/terraform/plugin/framework/providers#resources) function
+1. Follow the rest of the procedures in the [Provider Data Source or Resource Removal section](#provider-data-source-or-resource-removal).
+
+### Example Resource Renaming
```

Comment:
Same as above, I transcribed the original SDK deprecation code examples over, but is there a better place to return this `diag` when deprecating an entire resource?

Original example: https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/deprecations#provider-data-source-or-resource-rename